src/odemis/driver/simstreakcam.py

```diff
@@ -0,0 +1,537 @@
+
+        self._sync_event = event
+
+        if self._sync_event:
+            # softwareTrigger subscribes to onEvent method: if softwareTrigger.notify() called, onEvent method called
+            self._sync_event.subscribe(self)  # must have onEvent method
+            self._evtq = Queue.Queue()  # to be sure it's empty
+
+    @oneway
+    def onEvent(self):
+        """
+        Called by the Event when it is triggered  (e.g. self.softwareTrigger.notify()).
+        """
+        logging.debug("Event triggered to start a new synchronized acquisition.")
+        self._evtq.put(time.time())
+
+    # override
+    def updateMetadata(self, md):
+        """
+        Update the metadata.
+        """
+        super(ReadoutCamera, self).updateMetadata(md)
+        if model.MD_LENS_MAG in md:
+            self._updateWavelengthList()
+
+    def _mergeMetadata(self, md):
+        """
+        Create dict containing all metadata from the children readout camera, streak unit,
+        delay genereator and the metadata from the parent streak camera.
+        :return: merged metadata
+        """
+
+        md_devices = [self.parent._streakunit._metadata, self.parent._delaybox._metadata]
+
+        for md_dev in md_devices:
+            for key in md_dev.keys():
+                if key not in md.keys():
+                    md[key] = md_dev[key]
+                else:
+                    md[key] = md[key] + ", " + md_dev[key]
+                    # md[key].append(md_dev[key])  # TODO make nice  ", ".join(c)
+        return md
+
+    def terminate(self):
+        try:
+            self._stop()  # stop any acquisition
+        except Exception:
+            pass
+
+    def _start(self):
+        """
+        Start an acquisition.
+        """
+        if self._generator is not None:
+            logging.warning("Generator already running")
+            return
+        self._generator = util.RepeatingTimer(self.exposureTime.value,
+                                              self._generate,
+                                              "SimCam image generator")
+        self._generator.start()
+
+    def _stop(self):
+        """
+        Stop the acquisition.
+        """
+        if self._generator is not None:
+            self._generator.cancel()
+            self._generator = None
+
+    def _getNewImage(self):
+        """
+        Gets a new image from the list of fake images provided.
+        :return: (dataarray) image
+        """
+        # image counter to provide different images to live view
+        if not hasattr(self, "_img_counter"):
+            self._img_counter = 0  # initialize the image counter
+        self._img_counter = self._img_counter % len(self._img_list)
+        image = self._img_list[self._img_counter]
+        self._img_counter += 1
+
+        return image
+
+    def _generate(self):
+        """
+        Generates the fake output image based on the resolution.
+        """
+        gen_img = self._getNewImage()
```

Comment:
You could merge `getNewImage()` and `simulate()` into a single function, as they are always called together and essentially complete each other.